### PR TITLE
Allow config files to be placed per hostname in misc/user directory

### PR DESCRIPTION
**Pull request by tsteur**
We already allow users to have a custom config per hostname by putting them into `config/$hostname.config.ini.php`. On the contrary, custom logos are uploaded in `misc/user/$hostname/*`. 

It would make sense to have all files per hostname in `misc/user/$hostname`, including the config. This PR lets you put a config file into eg `misc/user/apache.matomo/config.ini.php` and then the custom config will be used for this host.

Old way is still supported, we just look in the two places now.

fyi @mattab

### CODE REVIEW

**Comment by tsteur** [comment]
FYI: We're running this already in production and works well

File: core/Config.php [hunk 1]
```diff
@@ -215,28 +229,42 @@ public static function getHostname()
...
     {
-        $hostConfig = self::getLocalConfigInfoForHostname($hostname);
+        $hostConfigs = self::getLocalConfigInfoForHostname($hostname);
+        $fileNames = '';
+
+        foreach ($hostConfigs as $hostConfig) {
+            if (count($hostConfigs) > 1
+                && $preferredPath
+                && strpos($hostConfig['path'], $preferredPath) === false) {
+                //
```
Comment (diosmosis) [inline core/Config.php]: Was there supposed to be a comment here?

Comment (tsteur) [reply core/Config.php]: will remove it

**Review by diosmosis — APPROVED** [review]
Wondering if this would be good to cover in a test somehow, otherwise looks good to me.

**Pull request merged**